libdnf/utils/smartcols/Line.hpp

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (C) 2018 Red Hat, Inc.
+ *
+ * Licensed under the GNU Lesser General Public License Version 2.1
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#ifndef LIBDNF_LINE_HPP
+#define LIBDNF_LINE_HPP
+
+#include "Cell.hpp"
+#include "Column.hpp"
+
+#include <memory>
+#include <libsmartcols/libsmartcols.h>
+
+class Table;
+
+class Line
+{
+    friend class Table;
+
+public:
+    explicit Line(struct libscols_line *line) : line(line) { scols_ref_line(line); };
+    Line(const Line &line) { this->line = line.line; };
```

Comment:
All Line objects point to the same line and call unref in destructor. May be `scols_copy_line()` is missing.